### ISSUE TITLE
Central and local directory mismatch for file?

### ISSUE THREAD

**Issue by DaveInCaz** [icsharpcode/SharpZipLib#399]
### Steps to reproduce
Sporadic, we're not sure how this happens.

### Actual behavior
We have seen a handful of cases where a ZIP created using FastZIP is corrupt. If you try to open it in WinZip, apparently the ZIP file header is readable, but extracting at least some files fails like this:

![1](https://user-images.githubusercontent.com/18285074/70177324-26d48480-16a8-11ea-94bf-724d8c1f9e82.png)

And running `ZipFile.TestArchive(true)` on such a file will return false.

I am looking for advice on how to track down this problem. Would be glad to provide any other relevant details that might help.

We have no other processes except FastZip calls to read/write the ZIP file contents. So I highly doubt anything secondary is corrupting the file.

### Version of SharpZipLib
1.2.0

### Obtained from (only keep the relevant lines)
- Package installed using NuGet

**Comment by piksel** [comment]
What is the real CRC32 of the file being compressed?
You can use this python snippet to show the CRC of the file:
```sh
cat header.json | python -c 'import sys;import zlib;print(hex(zlib.crc32(sys.stdin.read().encode())%(1<<32)))'
```

**Comment by piksel** [comment]
Also, could you provide the output of 
```
unzip -Z -v -h bad_file.zip
```
Or you can attach a bad file to this issue by dragging it to the comment textbox.

**Comment by DaveInCaz** [comment]
@piksel I'm afraid I do not have the original header.json file that it was attempting to compress. This zipping operation is done by an app which manages the uncompressed source files in a temp folder so they were not preserved.

I've attached the output of the unzip command.
[unzip.txt](https://github.com/icsharpcode/SharpZipLib/files/4041859/unzip.txt)
[Alpha-108-Test-Project.zip](https://github.com/icsharpcode/SharpZipLib/files/4041860/Alpha-108-Test-Project.zip)

I've also attached the ZIP file with the problem.

Thanks for your help looking into this.

**Comment by DaveInCaz** [comment]
Also, in https://github.com/icsharpcode/SharpZipLib/issues/398#issuecomment-561892704 it was mentioned that this issue is a variant of https://github.com/icsharpcode/SharpZipLib/issues/398. If that is definitely the case, would downgrading to version 1.1 be a valid workaround until the issue is permanently fixed?

**Comment by piksel** [comment]
That should probably suffice. I will take a look. If it is the same problem then that would indeed fix it, but I'm not so sure that it is. They seem to touch the same areas, but I don't see how this could be caused by the same bug.

**Comment by DaveInCaz** [comment]
Thanks. I'll try 1.1 though since I can't reproduce this at will it will be hard to know if that definitely has avoided the issue.

**Comment by piksel** [comment]
The only issue I can see in [the report](https://archivediagstore.z16.web.core.windows.net/reports/3OczwJ2oCkmYNl7IA6DibQ.html) is that the CRC for one file is different in the two headers. But without any clear way to reproduce this I'm going to close it. Hopefully it will be fixed in v1.3, but since we can't test it... 🤷‍♀️
